### ISSUE TITLE
`~` patch only versions breaks some fixed packages versions

### ISSUE THREAD

**Issue by adrien2p** [mikro-orm/mikro-orm#4448]
**Describe the bug**
I ve noticed that for example, the core package expect `@mikro-orm/knex` to install the version `~5.7.4` when using the core package with the same version. If you decide to have a fix version of the packages, then the installed knex package will be of a more recent version since `~` will allow any patch versions to be updated. The problem is that in `5.7.12` the`AbstractSqlConnection#getKnexOptions` is using `mergeConfig` which is not a method that is exported/exists in the core package version 5.7.4. Therefore, it breaks.

**Stack trace**
```
        const config = core_1.Utils.mergeConfig({
                                    ^

TypeError: core_1.Utils.mergeConfig is not a function
    at PostgreSqlConnection.getKnexOptions (/Users/adriendeperetti/Documents/02-personnel/01-projects/test-product-migration-cli/node_modules/@mikro-orm/knex/AbstractSqlConnection.js:118:37)
    at PostgreSqlConnection.createKnexClient (/Users/adriendeperetti/Documents/02-personnel/01-projects/test-product-migration-cli/node_modules/@mikro-orm/knex/AbstractSqlConnection.js:110:38)
    at PostgreSqlConnection.connect (/Users/adriendeperetti/Documents/02-personnel/01-projects/test-product-migration-cli/node_modules/@mikro-orm/postgresql/PostgreSqlConnection.js:12:28)
    at PostgreSqlDriver.connect (/Users/adriendeperetti/Documents/02-personnel/01-projects/test-product-migration-cli/node_modules/@mikro-orm/core/drivers/DatabaseDriver.js:53:31)
    at MikroORM.connect (/Users/adriendeperetti/Documents/02-personnel/01-projects/test-product-migration-cli/node_modules/@mikro-orm/core/MikroORM.js:70:46)
    at PostgreSqlMikroORM.init (/Users/adriendeperetti/Documents/02-personnel/01-projects/test-product-migration-cli/node_modules/@mikro-orm/core/MikroORM.js:39:23)
    at async createConnection (/Users/adriendeperetti/Documents/02-personnel/01-projects/test-product-migration-cli/node_modules/@medusajs/product/dist/utils/create-connection.js:8:17)
    at async runMigrations (/Users/adriendeperetti/Documents/02-personnel/01-projects/test-product-migration-cli/node_modules/@medusajs/product/dist/scripts/migration-up.js:40:17)
    at async /Users/adriendeperetti/Documents/02-personnel/01-projects/test-product-migration-cli/node_modules/@medusajs/product/dist/scripts/bin/run-migration-up.js:31:5

Node.js v18.13.0

```

**To Reproduce**
1. install @mikro-orm@5.7.4
2. init mikro orm

**Expected behavior**
The patch version shouldn't have introduced any breaking changes. I would say that the fact that the knex package use `mergeConfig` should have been part of a minor in order to not break earlier versions. what is your opinion on that?

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | 18.13.0 |
| typescript | ? |
| mikro-orm | 5.7.4 |
| your-driver | postgres |

**Comment by B4nan** [comment]
I guess we should lock the version instead of using `~` ranges. I am definitely not planning to deal with inter-package breaking changes, no plans to think about this sort of problems. So dont expect a fix for v5.7.4, you should just upgrade to latest, but I will try to deal with this for future releases.

**Comment by adrien2p** [comment]
I ll close this issue, but it might serve as an answer for other people running into the same issue
